lib/msal-common/src/error/InvalidGrantAuthError.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Constants } from "../utils/Constants";
+import { AuthError } from "./AuthError";
+
+export const InvalidGrantAuthErrorCodes = {
+    badToken: "bad_token",
+};
+
+/**
+ * InvalidGrantServerErrorMessage contains string constants used by error codes and messages returned by the server indicating interaction is required
+ */
+export const InvalidGrantServerErrorMessage = [
+    InvalidGrantAuthErrorCodes.badToken,
+];
+
+export const InvalidGrantAuthSubErrorMessage = ["bad_token"];
+
+const InvalidGrantAuthErrorMessages = {
+    [InvalidGrantAuthErrorCodes.badToken]:
+        "Server returned invalid_grant because of bad refresh token. Refresh token must be removed from cache.",
+};
+
+/**
+ * Invalid Grant errors defined by the SDK
+ * @deprecated Use InvalidGrantAuthErrorCodes instead
+ */
+export const InvalidGrantAuthErrorMessage = {
+    badToken: {
+        code: InvalidGrantAuthErrorCodes.badToken,
+        desc: InvalidGrantAuthErrorMessages[
+            InvalidGrantAuthErrorCodes.badToken
+        ],
+    },
+};
+
+/**
+ * Error thrown when invalid grant is used.
+ */
+export class InvalidGrantAuthError extends AuthError {
```

Comment:
Does this really need to be its own class? Why can't we continue to use InteractionRequiredAuthError/ServerError?

Reply:
Moved it back to InteractionRequired and added an exception for it in acquireTokenSilent's logic to determine if an error is silently resolvable